### PR TITLE
Decrypting group records

### PR DESCRIPTION
**Pull request by rosasck**
Attempt at creating this function

### CODE REVIEW

File: pdsClient/pdsClient.go [hunk 1]
```diff
@@ -426,6 +426,45 @@ func (c *E3dbPDSClient) DecryptRecord(ctx context.Context, record Record) (Recor
 	return record, err
 }
 
+// DecryptAK takes a record and an eak response, decrypts the membership key used for encrypting the record, and the public key
+// for the writer of the record. Then decrypts the record and returns data
+func (c *E3dbPDSClient) DecryptAK(ctx context.Context, record Record, akResponse *GetEAKResponse) (Record, error) {
```
Comment (galxy25) [inline pdsClient/pdsClient.go:431]: add group into the method name so its clear when this function would be used, and spell out AccessKey but you don't need to save on characters at the expense of readability
`DecryptGroupAcessKey`

Comment (galxy25) [reply pdsClient/pdsClient.go:431]: err, see how confusing this name is, I should have asked for

`DecryptGroupRecordWithGroupEncryptedAccessKey`

Comment (galxy25) [reply pdsClient/pdsClient.go:431]: Also `groupEncryptedAccessKey` vs `akResponse` (please ask yourself what meaning akResponse has to a random /first time reader of this method)

File: pdsClient/pdsClient.go [hunk 2]
```diff
@@ -426,6 +426,45 @@ func (c *E3dbPDSClient) DecryptRecord(ctx context.Context, record Record) (Recor
...
+	// otherwise attempt to decrypt the returned access key
+	rawEncryptionKey, err := e3dbClients.DecodeSymmetricKey(c.EncryptionKeys.Private.Material)
+	if err != nil {
+		return record, err
+	}
+	// Right now we only have one Wrapper, So this should return the Group Key
+	var rawGroupPrivateKey e3dbClients.SymmetricKey
+	var writerPublicKey string
+	for _, wrapper := range *akResponse.AccessKeyWrappers {
+
```
Comment (galxy25) [inline pdsClient/pdsClient.go:447]: remove extra newline

**Review by galxy25 — APPROVED** [review]
approved pending requested naming changes

**Comment by rosasck** [comment]
wierd error currently happening. Will debug after class/ tmr

**Comment by rosasck** [comment]
officially done, ready for pr

**Pull request merged**